### PR TITLE
New domain support

### PR DESCRIPTION
**Pull request by augustinas**
## Description
Refactors internal implementation, adds constructor for new HAT domain structure.

## Why
In order to support the new HAT domain structure the client needed to be updated. We decided that supplementing existing implementation with additional constructor was the safest backwards compatible approach.

## AC
Copy and paste from task

## Source
https://dswift.atlassian.net/browse/DAS-243

## Type
- [ ] Bug Fix
- [x] Feature Addition
- [x] Refactor
- [ ] HotFix

## Checklist
- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -1,32 +1,32 @@
-name-template: 'v$RESOLVED_VERSION'
-tag-template: 'v$RESOLVED_VERSION'
+name-template: "v$RESOLVED_VERSION"
```
Comment (whiteshadow-gr) [inline .github/release-drafter.yml:1]: I am not sure this is necessary plus now it's not consistent with the rest of our services

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,12 +17,34 @@ jobs:
...
+        uses: actions/setup-java@v2
         with:
+          distribution: adopt
           java-version: 11
           java-package: jdk
           architecture: x64
 
       - name: Clean and test
         run: sbt clean test
+
```
Comment (whiteshadow-gr) [inline .github/workflows/ci.yml:39]: Do we need the commented code?

**Review by whiteshadow-gr — APPROVED** [review]
Left 2 comments. But other than that looks good to me

File: .github/release-drafter.yml (hunk 1 above)
Comment (augustinas) [reply .github/release-drafter.yml:1]: @pperzyna ⬆️ ?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (augustinas) [reply .github/workflows/ci.yml:39]: @pperzyna ⬆️ ??

File: .github/release-drafter.yml (hunk 1 above)
Comment (pperzyna) [reply .github/release-drafter.yml:1]: It is just a visual studio YAML formatter, no difference - https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.quoted_strings

 It is the same as here: https://github.com/dataswift/HAT2.0/blob/main/.github/release-drafter.yml

File: .github/workflows/ci.yml (hunk 2 above)
Comment (pperzyna) [reply .github/workflows/ci.yml:39]: Leave it for now, I will add it letter as a separate PR

**Pull request merged**